### PR TITLE
fix: always consider user input when performing swap

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

**Comment by bee-worker** [comment]
## 🐝 PR preview in Swarm

**Preview URL:** https://bah5acgzamcu77rtbsimqeslgg2fsck5ppk5jdhthmzq5imox5d5r7ck3cohq.bzz.link
**Swarm Hash:** `60a9ffc6619219024966368b212baf7aba919e676661d431d7e8fb1f895b138f`
**Commit Hash:** 5198dbfd8bfce6d083687423a59c3e8b2de83351
**Commit Message:** fix: always consider user input when performing swap

File: src/pages/top-up/Swap.tsx [hunk 1]
```diff
@@ -52,28 +47,65 @@ export function Swap({ header }: Props): ReactElement {
...
-  }
+  // Set the initial xDAI to swap
+  useEffect(() => {
+    if (!balance) {
+      return
+    }
 
-  const optimalSwap = balance.dai.minusBaseUnits('1')
-  const lowAmountSwap = new DaiToken(balance.dai.toBigNumber.dividedToIntegerBy(2))
+    if (balance.dai.toDecimal.isGreaterThanOrEqualTo(DaiToken.fromDecimal('1').plusBaseUnits(MINIMUM_XDAI).toDecimal)) {
```
Comment (AuHau) [inline src/pages/top-up/Swap.tsx:62]: Can you please move out the `DaiToken.fromDecimal('1').plusBaseUnits(MINIMUM_XDAI).toDecimal)` into a separate variable for better readability?

File: src/pages/top-up/Swap.tsx [hunk 2]
```diff
@@ -52,28 +47,65 @@ export function Swap({ header }: Props): ReactElement {
...
+  // Set the xDAI to swap based on user input
+  useEffect(() => {
+    setError(null)
+    try {
+      if (userInputSwap) {
+        const dai = DaiToken.fromDecimal(userInputSwap)
+        setDaiToSwap(dai)
+
+        if (dai.toDecimal.lte(0)) {
+          setError('xDAI to swap must be a positive number')
```
Comment (AuHau) [inline src/pages/top-up/Swap.tsx:80]: Shouldn't there be also check for the minimal xDAI?

**Review by AuHau — APPROVED** [review]
Generally LGTM, some nitpicks and one question.

File: src/pages/top-up/Swap.tsx (hunk 2 above)
Comment (Cafe137) [reply src/pages/top-up/Swap.tsx:82]: That is part of the next useEffect:

```
if (daiAfterSwap.toDecimal.lt(MINIMUM_XDAI))
    setError(`Must keep at least ${MINIMUM_XDAI} xDAI after swap!`)
```

image

Comment (AuHau) [reply src/pages/top-up/Swap.tsx:82]: 👍

**Pull request merged**